baselayer/app/psa.py

```diff
@@ -4,55 +4,62 @@
 https://github.com/python-social-auth
 """
 
-from social_peewee.storage import (
-    database_proxy, BasePeeweeStorage,
-    PeeweeAssociationMixin, PeeweeCodeMixin,
-    PeeweeNonceMixin, PeeweePartialMixin, PeeweeUserMixin
-)
-
+from social_sqlalchemy.storage import (SQLAlchemyUserMixin,
+                                       SQLAlchemyAssociationMixin,
+                                       SQLAlchemyNonceMixin,
+                                       SQLAlchemyCodeMixin,
+                                       SQLAlchemyPartialMixin,
+                                       BaseSQLAlchemyStorage)
 from social_core.backends.google import GoogleOAuth2
 
-import peewee as pw
-from .models import BaseModel, User, db
-
-
-database_proxy.initialize(db)
+import sqlalchemy as sa
+from sqlalchemy.orm import relationship
+from .models import Base, User, DBSession#, db
 
 
-class UserSocialAuth(BaseModel, PeeweeUserMixin):
+class UserSocialAuth(Base, SQLAlchemyUserMixin):
     """
     This model is used by PSA to store whatever it needs during
     authentication, e.g. token expiration time, etc.
     """
-    user = pw.ForeignKeyField(User, related_name='social_auth')
+    uid = sa.Column(sa.String())
+    user_id = sa.Column(sa.ForeignKey('users.id', ondelete='CASCADE'))
+    user = relationship('User', backref='social_auth')
+    
+    def username_max_length():
+        return 255
+
+    @classmethod
+    def _session(cls):
+        return DBSession()
 
     @classmethod
     def user_model(cls):
         return User
 
 
-class TornadoPeeweeStorage(BasePeeweeStorage):
+class TornadoStorage(BaseSQLAlchemyStorage):
```

Comment:
We can probably use https://github.com/python-social-auth/social-app-tornado/blob/master/social_tornado/models.py directly.  I rewrote this class because we were using Peewee, and they didn't have that storage available.